src/main/java/io/temporal/internal/common/StartWorkflowExecutionParameters.java

```diff
@@ -348,14 +348,14 @@ public static StartWorkflowExecutionParameters fromWorkflowOptions(WorkflowOptio
       rp.setInitialIntervalInSeconds(roundUpToSeconds(retryOptions.getInitialInterval()));
       rp.setMaximumIntervalInSeconds(roundUpToSeconds(retryOptions.getMaximumInterval()));
       rp.setMaximumAttempts(retryOptions.getMaximumAttempts());
-      List<String> reasons = new ArrayList<>();
+      List<String> types = new ArrayList<>();
       // Use exception type name as the reason
-      List<Class<? extends Throwable>> doNotRetry = retryOptions.getDoNotRetry();
+      String[] doNotRetry = retryOptions.getDoNotRetry();
```

Comment:
What is reason for using string rather than types? Does previous usage fall under Illegal reflective access issues?

Reply:
This is to be compatible with other SDKs. An activity can be implemented in Go and return an error type which is not Java exception name.